# Abrimos los archivos a utilizar e importamos librerias

In [2]:
import pandas as pd


In [3]:
original=pd.read_csv('./accidentesoriginal.csv')
adicional=pd.read_csv('./este.csv')

Revisamos duplicados

In [4]:
def verDuplicados(df,col):
    total=df[df.duplicated(col)]
    
    return total

In [5]:
verDuplicados(original,'Unnamed: 0')

Empty DataFrame
Columns: [Unnamed: 0, fecha, HORA declarada, Ruta, OperadOR, flight_no, route, ac_type, registration, cn_ln, all_aboard, PASAJEROS A BORDO, crew_aboard, cantidad de fallecidos, passenger_fatalities, crew_fatalities, ground, summary]
Index: []

In [6]:
original

Unnamed: 0               fecha HORA declarada  \
0              0  September 17, 1908           1718   
1              1  September 07, 1909              ?   
2              2       July 12, 1912           0630   
3              3     August 06, 1913              ?   
4              4  September 09, 1913           1830   
...          ...                 ...            ...   
5003        5003      March 28, 2021           1835   
5004        5004        May 21, 2021           1800   
5005        5005       June 10, 2021           0800   
5006        5006       July 04, 2021          11:30   
5007        5007       July 06, 2021           1500   

                                    Ruta                         OperadOR  \
0                    Fort Myer, Virginia             Military - U.S. Army   
1                Juvisy-sur-Orge, France                                ?   
2              Atlantic City, New Jersey             Military - U.S. Navy   
3     Victoria, British Columbia, Canada                          Private   
4                     Over the North Sea           Military - German Navy   
...                                  ...                              ...   
5003                  Near Butte, Alaska                Soloy Helicopters   
5004                Near Kaduna, Nigeria    Military - Nigerian Air Force   
5005          Near Pyin Oo Lwin, Myanmar     Military - Myanmar Air Force   
5006          Patikul, Sulu, Philippines  Military - Philippine Air Force   
5007                      Palana, Russia    Kamchatka Aviation Enterprise   

     flight_no                         route                        ac_type  \
0            ?                 Demonstration               Wright Flyer III   
1            ?                      Air show                 Wright Byplane   
2            ?                   Test flight                      Dirigible   
3            ?                             ?               Curtiss seaplane   
4            ?                             ?         Zeppelin L-1 (airship)   
...        ...                           ...                            ...   
5003         ?           Sightseeing Charter    Eurocopter AS350B3 Ecureuil   
5004         ?                             ?  Beechcraft B300 King Air 350i   
5005         ?          Naypyidaw - Anisakan               Beechcraft 1900D   
5006         ?  Cagayan de Oro-Lumbia - Jolo       Lockheed C-130H Hercules   
5007       251        Petropavlovsk - Palana             Antonov An 26B-100   

     registration   cn_ln all_aboard PASAJEROS A BORDO crew_aboard  \
0               ?       1          2                 1           1   
1             SC1       ?          1                 0           1   
2               ?       ?          5                 0           5   
3               ?       ?          1                 0           1   
4               ?       ?         20                 ?           ?   
...           ...     ...        ...               ...         ...   
5003       N351SH    4598          6                 5           1   
5004       NAF203  FL-891         11                 7           4   
5005         4610   E-325         14                12           2   
5006         5125    5125         96                88           8   
5007     RA-26085   12310         28                22           6   

     cantidad de fallecidos passenger_fatalities crew_fatalities ground  \
0                         1                    1               0      0   
1                         1                    0               0      0   
2                         5                    0               5      0   
3                         1                    0               1      0   
4                        14                    ?               ?      0   
...                     ...                  ...             ...    ...   
5003                      5                    4               1      0   
5004                     11   

Exploramos el archivo

In [7]:
original.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5008 entries, 0 to 5007
Data columns (total 18 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   Unnamed: 0              5008 non-null   int64 
 1   fecha                   5008 non-null   object
 2   HORA declarada          5008 non-null   object
 3   Ruta                    5008 non-null   object
 4   OperadOR                5008 non-null   object
 5   flight_no               5008 non-null   object
 6   route                   5008 non-null   object
 7   ac_type                 5008 non-null   object
 8   registration            5008 non-null   object
 9   cn_ln                   5008 non-null   object
 10  all_aboard              5008 non-null   object
 11  PASAJEROS A BORDO       5008 non-null   object
 12  crew_aboard             5008 non-null   object
 13  cantidad de fallecidos  5008 non-null   object
 14  passenger_fatalities    5008 non-null   object
 15  crew

Podemos ver que el dataset contiene 5008 filas y 18 columnas.
Para visualizar los datos faltantes usamos isnull().sum()



In [8]:
original.isnull().sum()

Unnamed: 0                0
fecha                     0
HORA declarada            0
Ruta                      0
OperadOR                  0
flight_no                 0
route                     0
ac_type                   0
registration              0
cn_ln                     0
all_aboard                0
PASAJEROS A BORDO         0
crew_aboard               0
cantidad de fallecidos    0
passenger_fatalities      0
crew_fatalities           0
ground                    0
summary                   0
dtype: int64

En este archivo en particular los datos nulos o faltantes estan con un '?'.


In [29]:
original['HORA declarada'].value_counts()

?       1504
1500      35
1400      30
1100      30
1700      29
        ... 
2129       1
1002       1
1151       1
0909       1
1914       1
Name: HORA declarada, Length: 1217, dtype: int64

# Data Cleaning

En esta parte decidi dropear algunas columnas que no creo que me sirvan para mi dashboard.

In [9]:
df = original.drop(columns=['Unnamed: 0','HORA declarada', 'registration', 'cn_ln', 'registration', 'cn_ln', 'summary','PASAJEROS A BORDO','crew_aboard','passenger_fatalities','crew_fatalities','flight_no','route'])

In [10]:
df.fecha.value_counts()

August 31, 1988       4
September 11, 2001    4
February 23, 1945     3
December 28, 1946     3
August 27, 1992       3
                     ..
December 04, 1958     1
December 23, 1958     1
December 24, 1958     1
December 26, 1958     1
July 06, 2021         1
Name: fecha, Length: 4577, dtype: int64

In [11]:
df

fecha                                Ruta  \
0     September 17, 1908                 Fort Myer, Virginia   
1     September 07, 1909             Juvisy-sur-Orge, France   
2          July 12, 1912           Atlantic City, New Jersey   
3        August 06, 1913  Victoria, British Columbia, Canada   
4     September 09, 1913                  Over the North Sea   
...                  ...                                 ...   
5003      March 28, 2021                  Near Butte, Alaska   
5004        May 21, 2021                Near Kaduna, Nigeria   
5005       June 10, 2021          Near Pyin Oo Lwin, Myanmar   
5006       July 04, 2021          Patikul, Sulu, Philippines   
5007       July 06, 2021                      Palana, Russia   

                             OperadOR                        ac_type  \
0                Military - U.S. Army               Wright Flyer III   
1                                   ?                 Wright Byplane   
2                Military - U.S. Navy                      Dirigible   
3                             Private               Curtiss seaplane   
4              Military - German Navy         Zeppelin L-1 (airship)   
...                               ...                            ...   
5003                Soloy Helicopters    Eurocopter AS350B3 Ecureuil   
5004    Military - Nigerian Air Force  Beechcraft B300 King Air 350i   
5005     Military - Myanmar Air Force               Beechcraft 1900D   
5006  Military - Philippine Air Force       Lockheed C-130H Hercules   
5007    Kamchatka Aviation Enterprise             Antonov An 26B-100   

     all_aboard cantidad de fallecidos ground  
0             2                      1      0  
1             1                      1      0  
2             5                      5      0  
3             1                      1      0  
4            20                     14      0  
...         ...                    ...    ...  
5003          6                      5      0  
5004         11                     11      0  
5005         14                     12      0  
5006         96                     50      3  
5007         28                     28      0  

[5008 rows x 7 columns]

Creamos una funcion que recorre la columna del dataframe remplazando los '?' por un espacio en blanco

In [12]:
def RemplazaValoresColumna(dataframe,columna_in_str,to_replace_in_str,replaced_by_in_str):
     dataframe[columna_in_str]=dataframe[columna_in_str].astype('string').str.replace(to_replace_in_str,replaced_by_in_str)



In [17]:
RemplazaValoresColumna(df,'fecha','?','',)
RemplazaValoresColumna(df,'Ruta','?','',)
RemplazaValoresColumna(df,'ac_type','?','',)
RemplazaValoresColumna(df,'OperadOR','?','',)
RemplazaValoresColumna(df,'cantidad de fallecidos','?','',)
RemplazaValoresColumna(df,'ground','?','',)
RemplazaValoresColumna(df,'ac_type','?','',)



C:\Users\jank_\AppData\Local\Temp\ipykernel_23328\2804151166.py:2: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  dataframe[columna_in_str]=dataframe[columna_in_str].astype('string').str.replace(to_replace_in_str,replaced_by_in_str)


Cambiamos el formato de fecha de la columna

In [18]:
df['fecha']=df['fecha'] = pd.to_datetime(original.fecha, errors = 'coerce')

Guardamos el archivo listo para usar

In [21]:
df.to_csv('./archivofinal.csv', sep=',',index=False)

# Carga de tablas a Base de Datos MySql

In [24]:
file_1=pd.read_csv('./archivofinal.csv')


In [20]:
from sqlalchemy import create_engine


Realizamos la carga a traves de los datos de nuestro server introduciendo nuestro user y password

In [26]:
cadena_conexion= 'mysql+pymysql://root:123456789@localhost:3306/test'

conexion= create_engine(cadena_conexion)

def ImportaTablasAMySql(entry,nombre):  
    
    
    entry.to_sql(name=nombre, con=conexion)




Realizamos la carga

In [27]:
ImportaTablasAMySql(file_1,'Archivo1')

C:\Users\jank_\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\pandas\io\sql.py:1663: UserWarning: The provided table name 'Archivo1' is not found exactly as such in the database after writing the table, possibly due to case sensitivity issues. Consider using lower case table names.
  warnings.warn(msg, UserWarning)
